patches/api/0480-Add-ItemInHands-API-to-Allay.patch

```diff
@@ -0,0 +1,32 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Tamion <70228790+notTamion@users.noreply.github.com>
+Date: Tue, 14 May 2024 16:32:49 +0200
+Subject: [PATCH] Add ItemInHands API to Allay
+
+
+diff --git a/src/main/java/org/bukkit/entity/Allay.java b/src/main/java/org/bukkit/entity/Allay.java
+index e5e79202b97172efae31031c9c86808b9a268ce5..94ef12ced335076d8d0240ef47a9ba2bbbe62898 100644
+--- a/src/main/java/org/bukkit/entity/Allay.java
++++ b/src/main/java/org/bukkit/entity/Allay.java
+@@ -103,4 +103,21 @@ public interface Allay extends Creature, InventoryHolder {
+      */
+     @Nullable
+     public Location getJukebox();
++
++    // Paper start - Add ItemInHands API to Allay
++    /**
++     * Sets the item the allay is currently holding.
++     *
++     * @param item the item to hold
++     */
++    void setItemInHands(@Nullable org.bukkit.inventory.ItemStack item);
```

Comment:
I mean, `allay.getEquipment().setItemInMainHand(item);`

We can a) merge this into more entity api patch and b, probably just forward this call?
Or just not have this method in the first place, its how you'd set any other entities item in hand.

Reply:
oh welp, i opened this pr because yesterday someone on the discord was complaining they had to use nms to set the item and emily said she would open an issue (i guess she saw that method later aswell since she didn't open the issue yet) so after i couldn't find a method for it on the Inventory i just took their word for it and opened the pr lol. well ima close this then